test/e2e/suite_test.go

```diff
@@ -740,6 +741,11 @@ func createSingleDatacenterCluster(t *testing.T, ctx context.Context, namespace
 		}
 	}, polling.k8ssandraClusterStatus.timeout, polling.k8ssandraClusterStatus.interval)
 
+	t.Log("check that Cassandra DC was not restarted")
+	err = f.Client.Get(ctx, kcKey, k8ssandra)
+	require.NoError(err, "failed to get K8ssandraCluster in namespace %s", namespace)
+	require.Equal(dcGeneration, k8ssandra.Status.Datacenters["dc1"].Cassandra.ObservedGeneration)
+
```

Comment:
This is a bit futile given that the test uses Cassandra 3, I don't see any that uses 4. But I've verified manually that the test passes (and fails before the fix) with 4.0.6.

Reply:
We've had more flakiness in the tests with 4.0, possibly due to it requiring more resources 🤷 that's why all the tests moved to 3.11.
You could turn a light test (a single dc one) to use 4.0 and see how it goes maybe? The simplest one shouldn't deploy Stargate, which could be a good fit for your test.

Reply:
You can work around this by just checking if the flag is added even if Stargate isn't there. That would verify your change is working.

Reply:
In that case it should probably not be an e2e test. I'm trying to write an integration test right now, but I'm less familiar with that part of the codebase.

Reply:
+1 to an integration test

Reply:
works for me 👍